pkg/backend/refresh.go

```diff
@@ -2,11 +2,29 @@ package backend
 
 import (
 	"context"
+	"time"
 
+	"github.com/hashicorp/vault/sdk/framework"
 	"github.com/hashicorp/vault/sdk/logical"
 	"golang.org/x/oauth2"
 )
 
+func tokenOk2Reuse(tok *oauth2.Token, data *framework.FieldData) bool {
+	if !tok.Valid() {
+		return false
+	}
+	if data == nil {
+		return true
+	}
+	if minsecondsstr, ok := data.GetOk("minimum_seconds"); ok {
+		minseconds := minsecondsstr.(int)
+		if time.Until(tok.Expiry).Seconds() < float64(minseconds) {
```

Comment:
You may need a second branch here for tokens that don't provide an expiry.

Reply:
Possibly, although on the other hand it doesn't make much sense for somebody to use the minimum_seconds option if the token doesn't have an expiry.  It looks like that if there's no expiry, then tok.Expiry is 0.  time.Until() then should be a negative number, which means the function will return false.  I think that's a reasonable answer if no expiry is given and somebody asks for minimum_seconds.   I could document that behavior under the asterisk explaining the default.  On the other hand it would also be reasonable to say the token doesn't expire, so it should always be valid.  I'll go with whichever way you prefer, let me know.

Reply:
Actually, I added a test that re-read the "first" mock provider in refresh_test.go, the one that returns a random access token each time, no refresh token, and no expiry, and I realized that case is not relevant because of the extra check of `tok.RefreshToken == ""`.  It really doesn't make sense to have a refresh token if the access token never expires.  So I don't see any reason to change this.

Reply:
I take it back.  As soon as I added a call to tokenOk2Reuse in credsUpdateOperation as agreed, then this check did become relevant for the new test I mentioned.  I added the extra check for a zero Expiry.